### PR TITLE
general.fullname property is sometimes null

### PR DESCRIPTION
**Pull request by anzioka**
PR:
Some entities in `chronicle_user_apps` do not have `general.fullname` property. 

Screen Shot 2020-10-06 at 10 32 26 AM

### CODE REVIEW

File: src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java [hunk 1]
```diff
@@ -1151,6 +1151,14 @@ public boolean isKnownParticipant( UUID studyId, String participantId ) {
         return result;
     }
 
+    private String getESDataFirstValueOrNull( SetMultimap <FullQualifiedName, Object> entity, FullQualifiedName fqn ) {
```
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java:1154]: How about a generic name, `getFirstValueOrNull`?

File: src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java [hunk 2]
```diff
@@ -1151,6 +1151,14 @@ public boolean isKnownParticipant( UUID studyId, String participantId ) {
...
+        if ( entity.get( fqn ).isEmpty() ) {
+            return null;
+        }
+
+        return entity.get( fqn ).iterator().next().toString();
```
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java:1159]: Guava has a helper for this:

```
return Iterables.getFirst( entity.get( fqn ), null );
```

File: src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java [hunk 3]
```diff
@@ -1171,7 +1179,7 @@ public void refreshUserAppsFullNameValues() {
 
             // get entity key ids
             Set<String> fullNames = StreamUtil.stream( data )
-                    .map( entry -> entry.get( FULL_NAME_FQN ).iterator().next().toString() )
+                    .map( entry -> getFirstValueOrNull( entry, FULL_NAME_FQN ) )
```
Comment (katherinebernstein) [inline src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java:1182]: you probably want to filter out the nulls after this, i.e.
```
.map( ... )
.filter( value -> value != null)
.collect( ... )
```

(although someday when this is moved over to kotlin you can just use kotlin's built-in `.mapNotNull()` function :) )

Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java:1182]: Updated.

**Review by UnsungHero97 — APPROVED** [review]

**Review by katherinebernstein — APPROVED** [review]

**Pull request merged**